src/Core/Grid/Definition/Factory/CarrierGridDefinitionFactory.php

```diff
@@ -59,6 +61,40 @@ class CarrierGridDefinitionFactory extends AbstractGridDefinitionFactory
 
     public const GRID_ID = 'carrier';
 
+    /**
+     * @var string
+     */
+    protected $dbPrefix;
+
+    /**
+     * @var Connection
+     */
+    protected $connection;
+
+    /**
+     * @var bool
+     */
+    protected $showExternalModuleColumn;
+
+    /**
+     * @param HookDispatcherInterface $hookDispatcher
+     */
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher,
+        $dbPrefix,
+        Connection $connection
+    ) {
+        parent::__construct($hookDispatcher);
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+
+        $sql = 'SELECT count(external_module_name)
+                FROM ' . $this->dbPrefix . 'carrier
+                WHERE external_module_name <> ""';
```

Comment:
```suggestion
                WHERE active = 1 AND deleted = 0 AND external_module_name != ""';
```

Reply:
Sorry, you are right, I am dumb. :D

You did perfect, only `deleted` is correct. :-)

Reply:
> Sorry, you are right, I am dumb. :D
> 
> You did perfect, only `deleted` is correct. :-)

Ok Thanks!